Tests/Unit/EmogrifierTest.php

```diff
@@ -2121,4 +2118,48 @@ public function emogrifyMovesStyleElementFromHeadToBody()
             $result
         );
     }
+
+    /**
+     * Checks whether $html contains a $tagName tag with the $attribute attribute.
+     *
+     * @param string $html the HTML string we are searching in
+     * @param string $tagName the HTML tag we are looking for
+     * @param string $attribute the attribute we are looking for (with or even without a value)
+     * @return bool true if the HTML string contains the specified tag having the specified attribute
+     */
+    private function htmlStringContainsTagWithAttribute($html, $tagName, $attribute)
+    {
+        return preg_match('/<' . preg_quote($tagName, '/') . '[^>]+' . preg_quote($attribute, '/') . '/', $html) > 0;
+    }
+
+    /**
+     * @test
+     */
+    public function emogrifyPrefersInlineStyleOverCssBlockStyleForHtmlAttributesMapping()
+    {
+        $this->subject->setHtml(
+            '<html><head><style>p {width:1px}</style></head><body><p style="width:2px"></p></body></html>'
+        );
+        $this->subject->enableCssToHtmlMapping();
+
+        $result = $this->subject->emogrify();
+
+        self::assertTrue($this->htmlStringContainsTagWithAttribute($result, 'p', 'width="2"'));
+    }
+
+    /**
+     * @test
+     */
+    public function emogrifyCorrectsHtmlAttributesMappingWhenMultipleMatchingRulesAndLastRuleIsAuto()
+    {
+        $this->subject->setHtml(
+            '<html><head><style>p {width:1px}</style></head><body><p class="autoWidth"></p></body></html>'
+        );
+        $this->subject->setCss('p.autoWidth {width:auto}');
+        $this->subject->enableCssToHtmlMapping();
+
+        $result = $this->subject->emogrify();
+
+        self::assertFalse($this->htmlStringContainsTagWithAttribute($result, 'p', 'width='));
```

Comment:
> On the other hand, I think checking just for the attribute name without a value (width=) would be better here, because we should make sure that the width attribute is missing, not just that it does not have the value 1. What do you think?

Yes, I agree.

This would also allow us to change the private method to `assertHtmlStringContainsTagWithAttribute` (so that it already contains the `assert*` call in it).

Reply:
That was the first version of the private method, however here we need to make sure that there is no `$attribute` attribute for `$tagName`... and then `assertHtmlStringContainsTagWithAttribute` is not OK anymore. This is why I modified it to a `bool` function and used `assertTrue`/`assertFalse`. Am I missing something? (it's late here)

Reply:
Can't we then use a basic assertContains here instead?

Reply:
Yes, `assert*Not*Contains` should be enough.

Reply:
I was thinking of using an `assertContains` on the expected tag instead (by which we also test that the generated tag looks a certain way and not any other way).